arm-logic/2016-06-01/swagger/logic.json

```diff
@@ -0,0 +1,2302 @@
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/WorkflowRunAction"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Logic/locations/{location}/workflows/{workflowName}/validate": {
+      "post": {
+        "tags": [
+          "Workflows"
+        ],
+        "operationId": "Workflows_Validate",
+        "description": "Validates the workflow definition.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "name": "resourceGroupName",
+            "description": "The resource group name.",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "location",
+            "description": "The workflow location.",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "workflowName",
+            "description": "The workflow name.",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "name": "workflow",
+            "description": "The workflow definition.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/Workflow"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Resource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Gets or sets the resource id."
+        },
+        "name": {
+          "type": "string",
+          "description": "Gets the resource name."
+        },
+        "type": {
+          "type": "string",
+          "description": "Gets the resource type."
+        },
+        "location": {
+          "type": "string",
+          "description": "Gets or sets the resource location."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Gets or sets the resource tags."
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "SubResource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Gets or sets the resource id."
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "Object": {
```

Comment:
Is it possible to provide a more complete definition when parameters, etc. use this "Object" definition? If there are any properties that have constraints (e.g. type, required, etc), then there's no way for the consumer of the client to know this (and they would assume that any object they pass is valid). It also makes it difficult for the consumer to use any responses that return this object, since they have to look at the properties at run-time.